modules/creating-custom-links.adoc

```diff
@@ -13,56 +13,14 @@
 . Click *YAML* and edit the file:
 +
 ----
-// ConsoleLink is an extension for customizing OpenShift web console links.
-type ConsoleLink struct {
-	metav1.TypeMeta `json:",inline"`
-	// Standard object's metadata.
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              ConsoleLinkSpec `json:"spec"`
-}
-
-// ConsoleLinkSpec is the desired console link configuration.
-type ConsoleLinkSpec struct {
-	Link `json:",inline"`
-	// location determines which location in the console the link will be appended to.
-	Location ConsoleLinkLocation `json:"location"`
-	// applicationMenu holds information about section and icon used for the link in the
-	// application menu, and it is applicable only when location is set to ApplicationMenu.
-	//
-	// +optional
-	ApplicationMenu *ApplicationMenuSpec `json:"applicationMenu,omitempty"`
-}
-
-// ApplicationMenuSpec is the specification of the desired section and icon used for the link in the application menu.
-type ApplicationMenuSpec struct {
-	// section is the section of the application menu in which the link should appear.
-	Section string `json:"section"`
-	// imageUrl is the URL for the icon used in front of the link in the application menu.
-	// The URL must be an HTTPS URL or a Data URI. The image should be square and will be shown at 24x24 pixels.
-	// +optional
-	ImageURL string `json:"imageURL,omitempty"`
-}
-
-// ConsoleLinkLocationSelector is a set of possible menu targets to which a link may be appended.
-type ConsoleLinkLocation string
-
-const (
-	// HelpMenu indicates that the link should appear in the help menu in the console.
-	HelpMenu ConsoleLinkLocation = "HelpMenu"
-	// UserMenu indicates that the link should appear in the user menu in the console.
-	UserMenu ConsoleLinkLocation = "UserMenu"
-	// ApplicationMenu indicates that the link should appear inside the application menu of the console.
-	ApplicationMenu ConsoleLinkLocation = "ApplicationMenu"
-)
-
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
-type ConsoleLinkList struct {
-	metav1.TypeMeta `json:",inline"`
-	// Standard object's metadata.
-	metav1.ListMeta `json:"metadata"`
-	Items           []ConsoleLink `json:"items"`
-}
+apiVersion: console.openshift.io/v1
+kind: ConsoleLink
+metadata:
+  name: example
+spec:
+  href: 'https://www.example.com'
+  location: HelpMenu
```

Comment:
Do you feel a need to document all of the possible locations?  Currently there are 3:
- `HelpMenu` 
- `UserMenu`
- `ApplicationMenu`
There may be more in the future if we add them.

Reply:
Yeah, I can call those out. Thanks!

Reply:
There are quite a few ways to do this:

```yaml
apiVersion: console.openshift.io/v1
kind: ConsoleLink
metadata:
  name: namespaced-dashboard-link-for-all-namespaces
spec:
  href: 'https://www.example.com'
  location: NamespaceDashboard
  text: This appears in all namespaces 
```
vs a link that appears only for some namespaces:

```yaml
apiVersion: console.openshift.io/v1
kind: ConsoleLink
metadata:
  name: namespaced-dashboard-for-some-namespaces
spec:
  href: 'https://www.example.com'
  location: NamespaceDashboard
  # This text will appear in a box called "Laucher" under "namespace" or "project" in the console
  text: Fancy Namespace Dashboard LInk Thing
  namespaceDashboard:
    namespaces:
    # for these specific namespaces
    - my-namespace
    - your-namespace
    - other-namespace
```
vs this will cause a link to appear in the application menu:

```yaml
apiVersion: console.openshift.io/v1
kind: ConsoleLink
metadata:
  name: application-menu-link-1
spec:
  href: 'https://www.example.com'
  location: ApplicationMenu
  text: Link 1
  applicationMenu:
    section: My New Section
    # image that is 24x24 in size
    imageURL: https://via.placeholder.com/24
```

Reply:
Feel free to document as you see fit, I'm just sharing several examples.